helm/prometheus/templates/prometheus.yaml

```diff
@@ -60,9 +60,7 @@ spec:
   serviceMonitorSelector:
 {{ toYaml .Values.serviceMonitorsSelector | indent 4 }}
 {{- else }}
-  serviceMonitorSelector:
-    matchLabels:
-      prometheus: {{ .Release.Name }}
+  serviceMonitorSelector: {}
```

Comment:
The old labeling is still the better choice here I think, as we don't want these to possibly collide. cc @gianrubio